runtime/lua/vim/lsp/codelens.lua

```diff
@@ -299,12 +299,12 @@ function M.refresh(opts)
   local bufnr = opts.bufnr and resolve_bufnr(opts.bufnr)
   local buffers = bufnr and { bufnr }
     or vim.tbl_filter(api.nvim_buf_is_loaded, api.nvim_list_bufs())
-  local params = {
-    textDocument = util.make_text_document_params(),
-  }
 
   for _, buf in ipairs(buffers) do
     if not active_refreshes[buf] then
+      local params = {
```

Comment:
Isn't this change creating `params` for non-active instead?

A motivational example (e.g. an issue where a server is doing something different and causing the described problem) would help.

Reply:
Suppose that `hoge.lua` is open with `bufnr=1` and `fuga.vim` with `bufnr=2`, and the current buffer is `bufnr=1`.
If we call `vim.lsp.codelens.refresh()`, it tries to refresh all loaded buffers.

In the code before fix, `vim.lsp.util.make_text_document_params()` does not specify `bufnr`, so it creates `params` for the current buffer `hoge.lua`.

Then, try to run `vim.lsp.buf_request` for each buffer, but `params` for `hoge.lua` is always passed and executed.

BTW, `active_refreshes` remembers the buffers that are being refreshed, to avoid duplicate refresh operations.

Reply:
I see, I understand now.